lib/ansible/modules/cloud/docker/docker_node.py

```
@@ -0,0 +1,301 @@
+
+- name: Merge node labels and new labels
+  docker_node:
+    hostname: mynode
+    labels:
+      key: value
+    labels_state: merge
+
+- name: Remove all labels assigned to node
+  docker_node:
+    hostname: mynode
+    labels:
+    labels_state: replace
+
+- name: Remove node labels
+  docker_node:
+    hostname: mynode
+    labels_state: remove
+'''
+
+RETURN = '''
+node_facts:
+  description: Information about node after 'update' operation
+  returned: success
+  type: dict
+
+'''
+
+try:
+    from docker.errors import APIError
+except ImportError:
+    # missing docker-py handled in ansible.module_utils.docker_common
+    pass
+
+from ansible.module_utils.docker_common import (
+    DockerBaseClass,
+)
+
+from ansible.module_utils._text import to_native
+
+from ansible.module_utils.docker_swarm import AnsibleDockerSwarmClient
+
+
+class TaskParameters(DockerBaseClass):
+    def __init__(self, client):
+        super(TaskParameters, self).__init__()
+
+        # Spec
+        self.name = None
+        self.labels = None
+        self.labels_state = None
+
+        # Node
+        self.availability = None
+        self.role = None
+
+        for key, value in client.module.params.items():
+            setattr(self, key, value)
+
+
+class SwarmNodeManager(DockerBaseClass):
+
+    def __init__(self, client, results):
+
+        super(SwarmNodeManager, self).__init__()
+
+        self.client = client
+        self.results = results
+        self.check_mode = self.client.check_mode
+
+        self.client.fail_task_if_not_swarm_manager()
+
+        self.parameters = TaskParameters(client)
+
+        self.node_update()
+
+    def node_update(self):
+        if not (self.client.check_if_swarm_node(node_id=self.parameters.hostname)):
+            self.client.fail(msg="This node is not part of a swarm.")
+            return
+
+        try:
+            status_down = self.client.check_if_swarm_node_is_down()
+        except APIError:
+            return
```

I think you need some proper error handling here.

---

The `client.check_if_swarm_node_is_down()` calls the `get_node_inspect()` and there is extended handling of API exceptions. I've added here additional `fail()` call here instead of return just to be safe.

---

If no APIError can leave check_if_swarm_node_is_down, you should remove the try/except :)